### PR TITLE
fixes #143

### LINKED ISSUE #143: Reserve item should offer delivery methods/pickup locations
See: requests/8179402

"Request this Item" button and select checkbox display without delivery locations. If no other copies are loanable at Princeton (as is the example case), it should allow BD (and ILL?), but not Recall.

### PR DESCRIPTION
**Pull request by sdellis**
@eliotjordan or @tampakis 
I'm new to RuboCop. When running tests with `rake ci` I get the following offenses, which lead me to think that there are extra `end` statements in the `request_spec.rb` file. I've been through the spec and don't think this is the case, but I'm not sure how to resolve it. Obviously, the configuration tip seems promising, but I can't find what they mean by AllCops or where I can configure it. Any thoughts?

```
Offenses:

spec/models/requests/request_spec.rb:307:7: E: unexpected token kEND
(Using Ruby 2.1 parser; configure using TargetRubyVersion parameter, under AllCops)
      end
      ^^^
spec/models/requests/request_spec.rb:1283:1: E: unexpected token $end
(Using Ruby 2.1 parser; configure using TargetRubyVersion parameter, under AllCops)

```

### CODE REVIEW

File: spec/models/requests/request_spec.rb [hunk 1]
```diff
@@ -302,6 +302,10 @@
         expect(subject.requestable.first.location["code"]).to eq('sciresp')
       end
 
+      it "should be on reserve" do
+        expect(subject.requestable.first.on_reserve?.to be_truthy
```
Comment (eliotjordan) [inline spec/models/requests/request_spec.rb]: Missing a paren.

**Comment by eliotjordan** [comment]
Adding that missing paren fixes the problem. The real violation here is `E: unexpected token kEND`.

`(Using Ruby 2.1 parser; configure using TargetRubyVersion parameter, under AllCops)` can be ignored. It's annoying, but it's telling you that you are using a ruby 2.1 parser but you are using a different version of ruby. Likely 2.3. One reason this is set in this way is that ruby 2.3 includes an option to add `# frozen_string_literal: true` to each file in preparation for ruby 3.0. Rubocop requires that by default  if you're targeting 2.3 and above. If we're all using ruby >= 2.3 I can make a PR to change that and disable the `Style/FrozenStringLiteralComment` cop.

**Comment by sdellis** [comment]
@eliotjordan Thank you! I don't know how I missed that paren, but I did. Duh. 

Yes, I'm using 2.3 for this project but I don't know about others or other projects. The comment doesn't bother me, but in my Rubocop noobness (and familiarity with unhelpful JS errors) I thought that it might be related to the offense in some way.

**Comment by sdellis** [comment]
Ready for merge @kevinreiss

File: spec/models/requests/request_spec.rb [hunk 2]
```diff
@@ -280,6 +280,24 @@
...
+    let(:params) {
+      {
+        system_id: '8179402',
+        user: user
+      }
+    }
+    let(:request_with_items_on_reserve) { described_class.new(params) }
+    subject { request_with_items_on_reserve }
+
+    describe "#requestable" do
```
Comment (kevinreiss) [inline spec/models/requests/request_spec.rb]: I think adding a test for on_reserve = false will turn the coveralls readout to green.

**Comment by kevinreiss** [comment]
👍

**Pull request merged**